pandas/core/frame.py

```diff
@@ -1305,11 +1309,17 @@ def to_excel(self, excel_writer, sheet_name='Sheet1', na_rep='',
         """
         from pandas.io.excel import ExcelWriter
         if self.columns.nlevels > 1:
-            raise NotImplementedError("Writing as Excel with a MultiIndex is "
```

Comment:
this type of checking on the index should go about [here](https://github.com/pydata/pandas/blob/master/pandas/core/format.py#L1741). Then all of this code will be in one place.